### DISCUSSION TITLE
going through the haxe game tutorial and im stuck at part 8 due to my game crashing

### DISCUSSION THREAD

**Discussion by suskbak** [HaxeFlixel/flixel#2337] [Q&A] [answered]
after some experimenting i found out the bit of code that is doing this is the "else if (entity.name == "coin")
coins.add(new Coin(entity.x, entity.y));}"  

i tried searching it up but got nothing, and i dont really know what is wrong, can someone help?

here is the code
https://pastebin.com/AgpFFeeJ (playstate code)
https://pastebin.com/vHb9Kg4C (player code)
https://pastebin.com/g3u5k3uc (coin code)

help would be greatly appreciated

**Comment by TaylorAnderson** [comment, marked as answer]
When placeEntities gets run, your coin group hasnt been initialized yet.  Try moving coins = new FlxTypedGroup up to above the map.loadEntities call.

**Reply by suskbak** [reply]
sorry for the late response, your suggestion worked and it made it not crash when running the game, thank you.

**Reply by TaylorAnderson** [reply]
Could you mark my answer as the correct one so the issue is marked as resolved?

**Reply by suskbak** [reply]
yeah sorry i didnt know how to mark the forum as answered, its done now
